agent/command/s3_copy.go

```diff
@@ -220,11 +220,13 @@ func (c *s3copy) copyWithRetry(ctx context.Context,
 		}
 		if newPushLog.TaskId == "" {
 			logger.Task().Infof("noop, this version is currently in the process of trying to push, or has already succeeded in pushing the file: '%s/%s'", s3CopyFile.Destination.Bucket, s3CopyFile.Destination.Path)
-			return nil
+			timer.Reset(backoffCounter.Duration())
+			continue
 		}
 
 		if len(s3CopyFile.BuildVariants) > 0 && !utility.StringSliceContains(
 			s3CopyFile.BuildVariants, conf.BuildVariant.Name) {
+			timer.Reset(backoffCounter.Duration())
```

Comment:
Can we just set timer.Reset(0) at the top of the for loop? The backoff counter only seems necessary for the retryLoop, and that makes it less error prone if we ever need to add another if statement or etc.

Reply:
sure